cosmwasm/enclaves/shared/block-verifier/src/submit_block_signatures.rs

```diff
@@ -0,0 +1,207 @@
+        message_verifier.clear();
+    }
+
+    for tx in txs.tx.iter() {
+        // doing this a different way makes the code unreadable or requires creating a copy of
+        // tx. Feel free to change this if someone finds a better way
+        log::trace!(
+            "Got tx: {}",
+            if tx.len() < TX_THRESHOLD {
+                format!("{:?}", hex::encode(tx))
+            } else {
+                String::new()
+            }
+        );
+
+        let parsed_tx = unwrap_or_error!(tx_from_bytes(tx.as_slice()).map_err(|_| {
+            error!("Unable to parse tx bytes from proto");
+            sgx_status_t::SGX_ERROR_INVALID_PARAMETER
+        }));
+
+        message_verifier.append_wasm_from_tx(parsed_tx);
+    }
+
+    message_verifier.set_block_info(
+        header.header.height.value(),
+        header.header.time.unix_timestamp_nanos(),
+    );
+
+    #[cfg(feature = "random")]
+    {
+        let encrypted_random_slice =
+            slice::from_raw_parts(in_encrypted_random, in_encrypted_random_len as usize);
+
+        let decrypted = unwrap_or_error!(crate::verify::random::validate_encrypted_random(
+            encrypted_random_slice,
+            validator_set.hash(),
+            header.header.app_hash.as_bytes(),
+            header.header.height.value(),
+        ));
+
+        decrypted_random.copy_from_slice(&decrypted);
+    }
+
+    debug!(
+        "Done verifying block height: {:?}",
+        header.header.height.value()
+    );
+
+    sgx_status_t::SGX_SUCCESS
+}
+
+#[allow(clippy::too_many_arguments)]
+#[allow(unused_variables)]
+fn validate_inputs(
+    in_header: *const u8,
+    in_header_len: u32,
+    in_commit: *const u8,
+    in_commit_len: u32,
+    in_txs: *const u8,
+    in_txs_len: u32,
+    in_encrypted_random: *const u8,
+    in_encrypted_random_len: u32,
+    decrypted_random: &mut [u8; 32],
+) -> Result<(), sgx_status_t> {
+    validate_input_length!(in_header_len, "header", MAX_VARIABLE_LENGTH);
+    validate_input_length!(in_commit_len, "commit", MAX_VARIABLE_LENGTH);
+    validate_input_length!(in_txs_len, "txs", MAX_TXS_LENGTH);
+    validate_input_length!(
+        in_encrypted_random_len,
+        "encrypted random",
+        RANDOM_PROOF_LEN
+    );
+
+    validate_const_ptr!(
+        in_header,
+        in_header_len as usize,
+        Err(sgx_status_t::SGX_ERROR_INVALID_PARAMETER)
+    );
+    validate_const_ptr!(
+        in_commit,
+        in_commit_len as usize,
+        Err(sgx_status_t::SGX_ERROR_INVALID_PARAMETER)
+    );
+
+    #[cfg(feature = "random")]
+    validate_const_ptr!(
+        in_encrypted_random,
+        in_encrypted_random_len as usize,
+        Err(sgx_status_t::SGX_ERROR_INVALID_PARAMETER)
+    );
+
+    validate_mut_ptr!(
+        decrypted_random.as_mut_ptr(),
+        decrypted_random.len(),
+        Err(sgx_status_t::SGX_ERROR_INVALID_PARAMETER)
+    );
+
+    if in_txs_len != 0 && !in_txs.is_null() {
```

Comment:
You should also validate that if in_txs is_null then in_txs_len is 0 and vice versa

Reply:
Is already checked by the macro